src/pocof/Query.fs

```diff
@@ -3,92 +3,171 @@ namespace pocof
+        try
+            Some((x.Properties.Item prop).Value)
+        with
+        | _ -> None
+
+    let run (state: PocofData.InternalState) (entries: PocofData.Entry list) (props: Map<string, string>) =
+        let rec parseQuery (acc: Query list) (xs: string list) =
+            // TODO: state.QueryState.Operator is PocofData.NONE.
+            match xs with
+            | [] -> acc
+            | (x :: xs) ->
+                match xs with
+                | [] ->
+                    parseQuery
+                    <| if x.StartsWith ":" then
+                           acc
+                       else
+                           Normal x :: acc
+                    <| []
+                | y :: zs ->
+                    if x.StartsWith ":" then
+                        parseQuery
+                        <| Property(x.[1..].ToLower(), y) :: acc
+                        <| zs
+                    else
+                        parseQuery <| Normal x :: acc <| xs
+
+        let queries =
+            state.Query.Split [| ' ' |]
+            |> List.ofSeq
+            |> parseQuery []
+
+        // PocofDebug.logFile "./debug.log" queries
+
+        let values (o: PocofData.Entry) =
+            queries
+            |> List.fold
+                (fun acc x ->
+                    match x with
+                    | Property (k, v) ->
+                        let pk =
+                            match props.TryGetValue k with
+                            | true, v -> v
+                            | _ -> ""
+
+                        let p =
+                            match o with
+                            | PocofData.Dict (dct) -> dct /? pk
+                            | PocofData.Obj (o) -> o /?/ pk
+
+                        match p with
+                        | Some (pv) -> (pv, v) :: acc
+                        | None -> acc
+                    | Normal (v) ->
+                        match o with
+                        | PocofData.Dict (dct) -> (dct.Key, v) :: (dct.Value, v) :: acc
+                        | PocofData.Obj (o) -> (o, v) :: acc)
+                []
+            |> List.map (fun (s, v) -> (s.ToString(), v))
 
         let is q =
-            match s.QueryState.Matcher with
+            match state.QueryState.Matcher with
             | PocofData.EQ -> (==) << equalOpt
             | PocofData.LIKE -> (=*=) << likeOpt
             | PocofData.MATCH -> (=~=) << matchOpt
-            <| s.QueryState.CaseSensitive
+            <| state.QueryState.CaseSensitive
             <| q
 
-        let answer = if s.QueryState.Invert then not else id
-
-        let predicate (o: obj) =
-            let queries =
-                match s.QueryState.Operator with
-                | PocofData.NONE -> [ s.Query ]
-                | _ -> s.Query.Split(" ") |> List.ofSeq
-                |> List.map is
+        let answer =
+            if state.QueryState.Invert then
+                not
+            else
+                id
```

Comment:
Invert filter without a query results in an empty list.
It's not expected behavior. should list all entries in this case.